lib/linguist/heuristics.yml

```diff
@@ -195,6 +195,10 @@ disambiguations:
     #  : dependency
     # path/file.ext1 : some/path/../file.ext2
     pattern: '([\/\\].*:\s+.*\s\\$|: \\$|^[ %]:|^[\w\s\/\\.]+\w+\.\w+\s*:\s+[\w\s\/\\.]+\w+\.\w+)'
+- extensions: ['.dml']
+  rules:
+  - language: Device Modeling Language
+    pattern: '(^dml\s1.4;)'
```

Comment:
This will have no effect as Linguist doesn't know of any other language that uses the `.dml` extension. See my primary comment for more details.

Reply:
Ah, I was under the impression that if this regex didn't match, and given that there are no other languages in linguist that has a `.dml` extension, the language of the file would simply not be recognized.

Reply:
Nope. Linguist works like a funnel... everything goes in at the top and then works through each of the strategies in decreasing order of specificity until a single language is identified:

https://github.com/github-linguist/linguist/blob/5fad8d57605a914026a65b0e3ff6815d739944de/lib/linguist.rb#L51-L71

As Linguist would only know of the `.dml` extension with this language, it would stop the process there which as as you can see is before the heuristics strategy. It would also however apply this to all `.dml` files found on GitHub.

There is an exception to this: very generic extensions which we add to the [`generic.yml`](https://github.com/github-linguist/linguist/blob/main/lib/linguist/generic.yml) file. These are checked against the heuristics really early in the extension strategy.

Thinking about this further, I think `.dml` is a good candidate to add to `generic.yml` too which would then mean this heuristic _is_ used and will leave the other languages unaffected.

Please can you add this extension to the `generic.yml` file. This heuristic can also be improved (we don't need the capturing group and the tiny overhead it brings with it):

```suggestion
    pattern: '^dml\s1.4;'
```